dsa-solutions/lc-solutions/1000-1099/1008-Construct-Binary-Search-Tree-from-Preorder-Traversal.md

```diff
@@ -0,0 +1,131 @@
+id: construct-binary-search-tree-from-preorder-traversal
+title: Construct Binary Search Tree from Preorder Traversal
+level: medium
+sidebar_label: Construct BST from Preorder
+tags:
+  - Tree
+  - Depth-First Search (DFS)
+  - Binary Search Tree (BST)
+  - Java
+  - Python
+  - C++
+description: "This document provides solutions for the Construct Binary Search Tree from Preorder Traversal problem."
+---
+
+## Problem Statement
+
+Return the root node of a binary search tree that matches the given preorder traversal.
+
+(Recall that a binary search tree is a binary tree where for every node, any descendant of `node.left` has a value `< node.val`, and any descendant of `node.right` has a value `> node.val`. Also recall that a preorder traversal displays the value of the `node` first, then traverses `node.left`, then traverses `node.right`.)
+
+**Example 1:**
+
+Input: `preorder = [8,5,1,7,10,12]`
+
+Output: `TreeNode{val: 8, left: TreeNode{val: 5, left: TreeNode{val: 1, left: None, right: None}, right: TreeNode{val: 7, left: None, right: None}}, right: TreeNode{val: 10, left: None, right: TreeNode{val: 12, left: None, right: None}}}`
+
+Explanation:
+The constructed binary search tree from the preorder traversal is shown.
+
+![Example 1](images/example1.png)
```

Comment:
The image is not available. You can remove the image line or upload an image to use.